paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/SynchronizationActionBase.java

```diff
@@ -178,11 +178,22 @@ protected abstract void buildSink(
 
     protected FileStoreTable alterTableOptions(Identifier identifier, FileStoreTable table) {
         // doesn't support altering bucket here
-        Map<String, String> withoutBucket = new HashMap<>(tableConfig);
-        withoutBucket.remove(CoreOptions.BUCKET.key());
-
+        Map<String, String> dynamicOptions = new HashMap<>(tableConfig);
+        dynamicOptions.remove(CoreOptions.BUCKET.key());
+
+        Map<String, String> oldOptions = table.options();
+        // remove immutable options and options with equal values
+        dynamicOptions
+                .entrySet()
+                .removeIf(
+                        entry ->
+                                CoreOptions.getImmutableOptionKeys().contains(entry.getKey())
+                                        || Objects.equals(
+                                                oldOptions.get(entry.getKey()), entry.getValue()));
```

Comment:
`CoreOptions.getImmutableOptionKeys()` should be placed out of the `removeIf`.

Reply:
Thanks, done.